In [1]:
import sqlite3

In [2]:
con=sqlite3.connect('austin_texas.db')
cursor = con.cursor()

zip = cursor.execute("Select distinct value from ways_tags WHERE key like '%street%' and value like '%st';")
for i in cursor.fetchall():
    print i


(u'Canyon Circle East',)
(u'E 51st St',)
(u'Ledgeway East',)
(u'Rapid Springs Cove East',)
(u'Red River St',)
(u'W 10th St',)


In [4]:
# Explore the size of the database
con=sqlite3.connect('austin_texas.db')
cursor = con.cursor()

cursor.execute("Select Count(*) From nodes")
print cursor.fetchall()
cursor.execute("Select Count(*) From nodes_tags")
print cursor.fetchall()
cursor.execute("Select Count(*) From ways")
print cursor.fetchall()
cursor.execute("Select Count(*) From ways_tags")
print cursor.fetchall()
cursor.execute("Select Count(*) From ways_nodes")
print cursor.fetchall()

con.commit()

[(736203,)]
[(52703,)]
[(82267,)]
[(288044,)]
[(812361,)]


In [5]:
# Explore number of unique users

con=sqlite3.connect('austin_texas.db')
cursor = con.cursor()

cursor.execute("Select Count(Distinct uid) From nodes")
print cursor.fetchall()[0][0]
cursor.execute("SELECT COUNT(DISTINCT(e.uid)) FROM (SELECT uid FROM nodes UNION ALL SELECT uid FROM ways) e;")
print cursor.fetchall()[0][0]

381
431


In [6]:
# Explore number of nodes and ways

con=sqlite3.connect('austin_texas.db')
cursor = con.cursor()

cursor.execute("Select Count(Distinct id) From nodes")
print cursor.fetchall()[0][0]
cursor.execute("Select Count(Distinct id) From nodes_tags")
print cursor.fetchall()[0][0]
cursor.execute("Select Count(Distinct id) From ways")
print cursor.fetchall()[0][0]
cursor.execute("Select Count(Distinct id) From ways_tags")
print cursor.fetchall()[0][0]


736203
18271
82267
81526


There are 736203 unique nodes,and 18271 of them have tagss
There are 82267 unique ways, and 81526 of them have tags

In [7]:
# Explore number of chosen types of nodes, like cafes and shops

con=sqlite3.connect('austin_texas.db')
cursor = con.cursor()

cursor.execute("""Select count(distinct id) From nodes_tags Where value Like "%cafe%" """)
print cursor.fetchall()[0][0]

cursor.execute("""Select  count(distinct id) From nodes_tags Where value Like "%shop%" """)
print cursor.fetchall()[0][0]

76
21


In [9]:
# Explore number of ways on Longview St
con=sqlite3.connect('austin_texas.db')
cursor = con.cursor()

cursor.execute("""Select ways_tags.value, Count(Distinct id) From ways_tags Where ways_tags.value like "%Longview%" """)
print cursor.fetchall()[0][0]

Longview Street


##Improvements:
####1. Complete the tags for all the nodes.
There are only 1148 nodes out of 32714 have tags, so one of the main contribution could be add tags for all the other nodes.
####2. Specify all the street types.
In some cases it's very hard to detect a possible street. For example we can correct "Rio Grande St" into "Rio Grande  Street". But if the original value is "Rio Grande", then it's very hard to correct this and all similar ones programmatically.

####3. Benefits and problems:
The benefits are clear: the data will be more accurate and complete, so we could extract more valuable information out of it. The problem is they both require a lot of effort to do.